### PR TITLE
[Setup] Error alert refactor

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR refactors the `View.alert(isPresented:presentingError:)` modifier that has been used to display an alert for an error in a sample. The modifier was replaced with `View.errorAlert(presentingError:)`which presents when the passed error is not `nil`. This allowed the `isShowingAlert` Boolean variables, originally used to present the alert, to be removed from the samples, reducing redundant code. 

Overall, the changes in the samples was kept as minimal as possible and mostly involved updating the alert modifier and removing it's associated Boolean value. A few the samples required additional refactoring:

* Create load report
* Show device location with NMEA data sources

<details><summary>List of samples affected</summary><p>

* Add feature layers
* Add raster from file
* Analyze network with subnetwork trace
* Augment reality to show tabletop scene
* Browse building floors
* Create and save KML
* Create load report
* Create mobile geodatabase
* Display annotation
* Display dimensions
* Display map from mobile map package
* Display points using clustering feature reduction
* Download vector title to local cache
* Filter features in scene
* Find closest facility from point
* Find closest facility to multiple points
* Find route
* Find route around barriers
* Find route in mobile map package
* Generate offline map
* Identify graphics offline
* Identify KML features
* Identify layer features
* Identify raster cell
* Manage operational layers
* Navigate route
* Play KML tour
* Query feature table
* Select features in feature layer
* Set up location driven geotriggers
* Set up visibility of subtype sublayer
* Show device location
* Show device location history
* Show device location with NMEA data sources
* Show labels on layer
* Show mobile map package expiration date
* Show results of spatial relationships
* Style symbols from mobile style file

</p></details>

## Linked Issue(s)

- `swift/issues/4752`

## How To Test

**Error Alert:**
<details><summary>Test Code</summary><p>

```swift
struct DisplayMapView: View {
    /// A map with imagery basemap.
    @State private var map = Map(basemapStyle: .arcGISImagery)
    
    /// The error shown in the error alert.
    @State private var error: Error?
    
    var body: some View {
        // Creates a map view to display the map.
        MapView(map: map)
            .onSingleTapGesture { _, _ in
                do {
                    throw TestError.someError
                } catch {
                    self.error = error
                }
            }
            .onLongPressGesture { _, _ in
                do {
                    throw TestError.errorWithDescription
                } catch {
                    self.error = error
                }
            }
            .errorAlert(presentingError: $error)
    }
}

enum TestError: LocalizedError {
    case someError, errorWithDescription
    
    var errorDescription: String? {
        switch self {
        case .someError:
            return nil
        case .errorWithDescription:
            return "Some description."
        }
    }
}
```

</p></details>

* Copy and paste the test code above into the `Display map` sample.
* In the sample, tap on the map to see an error alert from a thrown error.
* Tap and hold to see an error alert for an error with a description.

**Samples:**

* Ensure all the samples updated work as expected.
* **Note:** Pay careful attention to cases were the state of the `error` is used outside of it being set or passed to the `errorAlert`. The alert now sets the `error` to `nil` when it is dismissed, which can cause unintended side effects if the `error` is used elsewhere.

### CODE REVIEW

File: Shared/Supporting Files/Extensions/View+ErrorAlert.swift [hunk 1]
```diff
@@ -15,16 +15,14 @@
...
-    ///   - isPresented: A binding to a Boolean value that determines whether
-    ///   to present the alert. When the user taps one of the alert’s actions,
-    ///   the system sets this value to false and dismisses the alert.
-    ///   - error: An error to be shown in the alert.
-    func alert(isPresented: Binding<Bool>, presentingError error: Error?) -> some View {
-        alert("Error", isPresented: isPresented, presenting: error) { _ in
-            EmptyView()
+    /// Presents an alert with a given error's `localizedDescription` as the message.
+    /// - Parameter error: A binding to an optional `Error` that will present the alert when
+    /// it is not `nil`. When the user taps "OK", this value to is set to `nil` and the alert is dismissed.
```
Comment (yo1995) [inline Shared/Supporting Files/Extensions/View+ErrorAlert.swift:20]:
```suggestion
    /// it is not `nil`. When the user taps "OK", this value is set to `nil` and the alert is dismissed.
```

**Review by yo1995 — COMMENTED** [review]
Looks good overall

**Review by mhdostal — DISMISSED** [review]
Looks great. Very nice!

**Review by mhdostal — APPROVED** [review]

File: Shared/Supporting Files/Extensions/View+ErrorAlert.swift [hunk 2]
```diff
@@ -17,9 +17,19 @@ import SwiftUI
...
     /// Presents an alert with a given error's `localizedDescription` as the message.
     /// - Parameter error: A binding to an optional `Error` that will present the alert when
-    /// it is not `nil`. When the user taps "OK", this value is set to `nil` and the alert is dismissed.
+    /// it is not `nil`. When the user taps "OK", this value is set to `nil`, and the alert is dismissed.
+    ///
+    /// If the given `error` is a `CancellationError`, it is ignored, and the alert is not presented.
     func errorAlert(presentingError error: Binding<Error?>) -> some View {
-        alert("Error", isPresented: .constant(error.wrappedValue != nil), presenting: error.wrappedValue) { _ in
+        /// A binding to a Boolean value indicating whether to present the alert.
+        let isPresented: Binding<Bool>
```
Comment (yo1995) [inline Shared/Supporting Files/Extensions/View+ErrorAlert.swift:25]: Arguably the conditions can be assigned to the boolean, but I like the readability by putting it into individual lines. :+1:

**Review by yo1995 — APPROVED** [review]

**Pull request merged**